smdebug/core/access_layer/s3handler.py

```diff
@@ -260,3 +267,176 @@ def close_client(self):
+            ) as botocore_error:
+                if isinstance(botocore_error, ClientError):
+                    if botocore_error.response["Error"]["Code"] != "AccessDenied":
+                        self.logger.warning(
+                            f"Unable to read tensor "
+                            f"from object {object_request.bucket}/{str(object_request.key)}"
+                            f": {str(botocore_error)}"
+                        )
+                    else:
+                        raise botocore_error
+                else:
+                    raise botocore_error
+            except Exception as e:
+                self.logger.warning(str(e))
+                msg = (
+                    "Unable to read tensor from object "
+                    + str(object_request.bucket)
+                    + "/"
+                    + str(object_request.key)
+                )
+                if object_request.length is not None:
+                    msg += (
+                        " from bytes "
+                        + str(object_request.start)
+                        + "-"
+                        + str(object_request.start + object_request.length - 1)
+                    )
+                self.logger.warning(msg)
+            count += 1
+        self.logger.warning("Unable to fetch file " + str(object_request.key))
+        return None
+
+    def get_objects(self, object_requests):
+        if type(object_requests) != list:
+            raise TypeError("get_objects accepts a list of ReadObjectRequest objects.")
+        data = []
+        for object_request in object_requests:
+            data.append(self.get_object(object_request))
+        return data
+
+    def delete_prefix(self, path, delete_request=None):
+        if path is not None and delete_request is not None:
+            raise ValueError("Only one of path or delete_request can be passed")
+        elif path is not None:
+            on_s3, bucket, prefix = is_s3(path)
+            if on_s3 is False:
+                raise ValueError("Given path is not an S3 location")
+            delete_requests = [DeleteRequest(bucket, prefix)]
+            self.delete_prefixes(delete_requests)
+        elif delete_request is not None:
+            self.delete_prefixes([delete_request])
+
+    def delete_prefixes(self, delete_requests):
+        if delete_requests is not None and len(delete_requests) == 0:
+            return
+        for delete_request in delete_requests:
+            list_request = ListRequest(Bucket=delete_request.bucket, Prefix=delete_request.prefix)
+            keys = self.list_prefix(list_request)
+            if len(keys):
+                delete_dict = {}
+                delete_dict["Objects"] = []
+                for key in keys:
+                    delete_dict["Objects"].append({"Key": key})
+                self.client.delete_objects(Bucket=delete_request.bucket, Delete=delete_dict)
+
+    def close_client(self):
```

Comment:
why the pass method? 
Should you close client here. and also create a destructor which closes the client.

Reply:
I added it here to make the Handlers consistent for testing/benchmarking. Might remove it. For sync, there's nothing to do in close. Async, we have something to close.